x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/include_exclude_options.tsx

```diff
@@ -0,0 +1,309 @@
+      [operation]: selectedOptions,
+    };
+    setIncludeExcludeSelectedOptions(options);
+    const terms = selectedOptions.map((option) => {
+      if (!Number.isNaN(Number(option.label))) {
+        return Number(option.label);
+      }
+      return option.label;
+    });
+    const param = `${operation}IsRegex`;
+    updateParams(operation, terms, param, false);
+  };
+
+  const onCreateOption = (
+    searchValue: string,
+    flattenedOptions: IncludeExcludeOptions[] = [],
+    operation: 'include' | 'exclude'
+  ) => {
+    const newOption = {
+      label: searchValue,
+    };
+
+    let includeExcludeOptions = [];
+
+    const includeORExcludeSelectedOptions = includeExcludeSelectedOptions[operation] ?? [];
+    includeExcludeOptions = [...includeORExcludeSelectedOptions, newOption];
+    const options = {
+      ...includeExcludeSelectedOptions,
+      [operation]: includeExcludeOptions,
+    };
+    setIncludeExcludeSelectedOptions(options);
+
+    const terms = includeExcludeOptions.map((option) => {
+      if (!Number.isNaN(Number(option.label))) {
+        return Number(option.label);
+      }
+      return option.label;
+    });
+    const param = `${operation}IsRegex`;
+    updateParams(operation, terms, param, false);
+  };
+
+  return (
+    <>
+      <EuiSpacer size="m" />
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.indexPattern.terms.include', {
+          defaultMessage: 'Include terms',
+        })}
+        display="rowCompressed"
+        fullWidth
+        labelAppend={
+          !isNumberField ? (
+            <EuiSwitch
+              label={
+                <EuiText size="xs">
+                  {i18n.translate('xpack.lens.indexPattern.terms.addRegex', {
+                    defaultMessage: 'Use a regex instead',
+                  })}
+                </EuiText>
+              }
+              data-test-subj="lens-include-terms-regex-switch"
+              compressed
+              checked={isRegexUsed.include}
+              onChange={(e) => {
+                const value = e.target.checked;
+                setIsRegexUsed({
+                  ...isRegexUsed,
+                  include: value,
+                });
+                setRegex({
+                  ...regex,
+                  include: '',
+                });
+                setIncludeExcludeSelectedOptions(getTermsIncludeExcludeOptions([], exclude));
+                updateParams('include', [], 'includeIsRegex', value);
+              }}
+            />
+          ) : null
+        }
+      >
+        {isRegexUsed.include ? (
+          <EuiFieldText
+            placeholder={i18n.translate(
+              'xpack.lens.indexPattern.terms.includeExcludePatternPlaceholder',
+              {
+                defaultMessage: 'Use a regular expression i.e. .*sport.*',
+              }
+            )}
+            data-test-subj="lens-include-terms-regex-input"
+            value={regex.include}
+            onChange={(e) => {
```

Comment:
This should be debounced using our `useDebouncedValue` helper - otherwise typing is very laggy